source/server/listener_manager_impl.cc

```diff
@@ -435,6 +435,10 @@ bool ListenerManagerImpl::addOrUpdateListenerInternal(
                                        workers_started_, hash, server_.options().concurrency());
   }
 
+#ifdef ENVOY_ENABLE_QUIC
+  new_listener->setQuicStatNames(quic_stat_names_);
```

Comment:
As above I think ListenerImpl can just grab the `QuicStatNames&` object from the manager.

Reply:
Or...if that involves changes to the base class you'd prefer not to do, you can just pass in the stats to the constructor rather than setting it after construction.

That just means you need to ifdef the arg-list.

Reply:
Fortunately ListenerImpl takes ListenerManagerImpl in its constructors. So I added getter in ListenerManagerImpl and let ListenerImpl grab it during constructions.